pkg/remote/trans/netpollmux/client_handler_test.go

```diff
@@ -108,3 +110,214 @@ func TestCliTransHandler(t *testing.T) {
+				l := in.ReadableLen()
+				test.Assert(t, l == 3*codec.Size32+len(s))
+				in.Skip(3 * codec.Size32)
+				got, err := in.ReadString(len(s))
+				test.Assert(t, err == nil, err)
+				test.Assert(t, got == s, got, s)
+				return err
+			},
+		},
+		ConnPool: NewMuxConnPool(1),
+	}
+
+	handler, err := NewCliTransHandlerFactory().NewTransHandler(opt)
+	test.Assert(t, err == nil)
+
+	ctx := context.Background()
+
+	var isWriteBufFlushed bool
+
+	buf := netpoll.NewLinkBuffer(1024)
+	npconn := &MockNetpollConn{
+		ReaderFunc: func() (r netpoll.Reader) {
+			return buf
+		},
+		WriterFunc: func() (r netpoll.Writer) {
+			isWriteBufFlushed = true
+			return buf
+		},
+	}
+	conn := newMuxCliConn(npconn)
+
+	ri := newMockRPCInfo()
+	msg := &MockMessage{
+		RPCInfoFunc: func() rpcinfo.RPCInfo {
+			return ri
+		},
+		ServiceInfoFunc: func() *serviceinfo.ServiceInfo {
+			return &serviceinfo.ServiceInfo{}
+		},
+	}
+
+	// 2. test
+	err = handler.Write(ctx, conn, msg)
+	// check err not nil
+	test.Assert(t, err != nil, err)
+	test.Assert(t, !isWriteBufFlushed)
+}
+
+// TestWriteOneWayMethod test client_handler write oneway method
+func TestWriteOneWayMethod(t *testing.T) {
+	// 1. prepare mock data
+	s := "hello world"
+	opt := &remote.ClientOption{
+		Codec: &MockCodec{
+			EncodeFunc: func(ctx context.Context, msg remote.Message, out remote.ByteBuffer) error {
+				r := mockHeader(msg.RPCInfo().Invocation().SeqID(), s)
+				n, err := out.WriteBinary(r.Bytes())
+				test.Assert(t, err == nil, err)
+				test.Assert(t, n == r.Len(), n, r.Len())
+				return err
+			},
+			DecodeFunc: func(ctx context.Context, msg remote.Message, in remote.ByteBuffer) error {
+				l := in.ReadableLen()
+				test.Assert(t, l == 3*codec.Size32+len(s))
+				in.Skip(3 * codec.Size32)
+				got, err := in.ReadString(len(s))
+				test.Assert(t, err == nil, err)
+				test.Assert(t, got == s, got, s)
+				return err
+			},
+		},
+		ConnPool: NewMuxConnPool(1),
+	}
+
+	handler, err := NewCliTransHandlerFactory().NewTransHandler(opt)
+	test.Assert(t, err == nil)
+
+	ctx := context.Background()
+	ri := newMockRPCInfo()
+
+	buf := netpoll.NewLinkBuffer(1024)
+	npconn := &MockNetpollConn{
+		ReaderFunc: func() (r netpoll.Reader) {
+			return buf
+		},
+		WriterFunc: func() (r netpoll.Writer) {
+			return buf
+		},
+	}
+	conn := newMuxCliConn(npconn)
+
+	oneWayMsg := &MockMessage{
+		RPCInfoFunc: func() rpcinfo.RPCInfo {
+			return ri
+		},
+		ServiceInfoFunc: func() *serviceinfo.ServiceInfo {
+			return &serviceinfo.ServiceInfo{
+				Methods: map[string]serviceinfo.MethodInfo{
+					"method": serviceinfo.NewMethodInfo(nil, nil, nil, true),
+				},
+			}
+		},
+	}
+	err = handler.Write(ctx, conn, oneWayMsg)
+	test.Assert(t, err == nil, err)
+}
+
+// TestReadTimeout test client_handler read timeout because of no execute OnRequest
+func TestReadTimeout(t *testing.T) {
+	s := "hello world"
+	opt := &remote.ClientOption{
```

Comment:
there are too many repeated codes in creating a client option, can we create a helper function to make it?

Reply:
It has been modified, please help code review again